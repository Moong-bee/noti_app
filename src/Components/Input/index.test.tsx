import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'

import { Input } from './index'

describe('<Input />', () => {
  it('올바른 랜더링 - placeholder', () => {
    const { container } = render(<Input placeholder="default placeholder" />)

    const input = screen.getByPlaceholderText('default placeholder')
    expect(input).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  it('올바른 랜더링 - value', () => {
    render(<Input value="default value" />)

    const input = screen.getByDisplayValue('default value')
    expect(input).toBeInTheDocument()
  })

  it('input 값 변경', () => {
    render(<Input placeholder="default placeholder" />)

    const input = screen.getByPlaceholderText('default placeholder') as HTMLInputElement
    fireEvent.change(input, { target: { value: 'default value' } })
    expect(input.value).toBe('default value')
  })
})
