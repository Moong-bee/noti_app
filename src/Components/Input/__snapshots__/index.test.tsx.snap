// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> 올바른 랜더링 - placeholder 1`] = `
<div>
  <input
    class="sc-bdnxRM lanArR"
    placeholder="default placeholder"
    value=""
  />
</div>
`;
